### PR TITLE
Modernize type hints

### PR DESCRIPTION
**Pull request by robertwb**
Convert from Python2 style comment type hints to true Python 3 annotations.

This was done with com2ann.

This PR only covers those files that have not been edited recently, and is split by top-level directory to reduce the chance and cost of conflicts.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @tvalentyn

**Comment by tvalentyn** [comment]
Run Python 3.8 PostCommit

**Comment by tvalentyn** [comment]
postcommits will be here: https://github.com/apache/beam/actions/runs/9848030460

there were linter failures as well - have you also tried importing this and running internal tests?

**Comment by robertwb** [comment]
Good call. Internal tests look good on import.

**Comment by robertwb** [comment]
Fixed the lint issue.

File: sdks/python/apache_beam/runners/direct/executor.py [hunk 1]
```diff
@@ -84,12 +84,12 @@ def _update_name(self, task=None):
...
           self._index, name, 'executing' if task else 'idle')
 
-    def _get_task_or_none(self) -> Optional[_ExecutorService.CallableTask]:
+    def _get_task_or_none(self) -> Optional['_ExecutorService.CallableTask']:
       try:
         # Do not block indefinitely, otherwise we may not act for a requested
         # shutdown.
         return self.queue.get(
-            timeout=_ExecutorService._ExecutorServiceWorker.TIMEOUT)
+            timeout='_ExecutorService._ExecutorServiceWorker.TIMEOUT')
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/direct/executor.py]: Is this an intentional change?

Comment (tvalentyn) [reply sdks/python/apache_beam/runners/direct/executor.py]: nvm, it's only in an outdated commit.

File: sdks/python/apache_beam/runners/direct/test_stream_impl.py [hunk 2]
```diff
@@ -309,8 +310,8 @@ def is_alive():
       return not (shutdown_requested or evaluation_context.shutdown_requested)
 
     # The shared queue that allows the producer and consumer to communicate.
-    channel: Queue[Union[test_stream.Event,
-                         _EndOfStream]] = Queue()  # noqa: F821
+    channel: 'Queue[Union[test_stream.Event, _EndOfStream]]' = (  # noqa: F821
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/direct/test_stream_impl.py:313]: does the entire hint need to be quoted or only '_EndOfStream' ?

**Review by tvalentyn — APPROVED** [review]
I looked at manual commits and spot-checked the autogenerated content, but there is a lot of changes and I could have missed something. What is the worst thing that can go wrong here? Inferring incorrect coder because there is a typo in a typehint? Do we need to be more careful and split it up into smaller PRs ?

File: sdks/python/apache_beam/runners/direct/test_stream_impl.py (hunk 2 above)
Comment (robertwb) [reply sdks/python/apache_beam/runners/direct/test_stream_impl.py:313]: Also test_stream.Event. Given the noqa, it seems there were issues with this before.

**Comment by robertwb** [comment]
Yes, looking at the manual changes and spot checking the auto generated should be sufficient. 

The worst thing is failure to import because of some circular/forward definition. I think I fixed all of those (even if things aren't imported by default, the docbuilder import most modules.) As you mention, some coders are inferred from type hints, but that should fail quickly because of actually being incorrect (and I didn't run into any of this in all the tests).

**Comment by robertwb** [comment]
Also, if there are issues we should be able to revert individual commits rather than roll the whole thing back. 

I'll wait 'till the release is cut before merging this to give it extra time to bake.

**Pull request merged**